chromadb/test/property/test_add.py

```diff
@@ -179,22 +185,29 @@ def test_add_large(
         embedding_function=collection.embedding_function,
     )
     normalized_record_set = invariants.wrap_all(record_set)
-    initial_version = coll.get_model()["version"]
+    initial_version = cast(int, coll.get_model()["version"])
 
+    ids = []
     for batch in create_batches(
         api=client,
         ids=cast(List[str], record_set["ids"]),
         embeddings=cast(Embeddings, record_set["embeddings"]),
         metadatas=cast(Metadatas, record_set["metadatas"]),
         documents=cast(List[str], record_set["documents"]),
     ):
-        coll.add(*batch)
+        result = coll.add(*batch)
+        if result["ids"] is not None:
+            ids.extend(result["ids"])
+
+    n_ids = (
+        len(normalized_record_set["ids"])
+        if normalized_record_set["ids"] is not None
+        else 0
+    )
+    if n_ids == 0:
+        normalized_record_set["ids"] = ids
 
-    if (
-        not NOT_CLUSTER_ONLY
-        and should_compact
-        and len(normalized_record_set["ids"]) > 10
-    ):
+    if not NOT_CLUSTER_ONLY and should_compact and n_ids > 10:
```

Comment:
Let's create an issue to fold the `not NOT_CLUSTER_ONLY and should_compact` logic into `wait_for_version_increase` everywhere.

Reply:
https://github.com/chroma-core/chroma/issues/2738 did this yesterday